instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/Network.kt

```diff
@@ -12,13 +22,60 @@ suspend fun BaseCoreNoCleanUpTest.withoutInternet(block: suspend () -> Unit) {
 }
 
 suspend fun withoutWifiAndMobileData(block: suspend () -> Unit) {
-    val uiAutomation = InstrumentationRegistry.getInstrumentation().uiAutomation
+    val reachability = ReachabilityFactory.reachability(null)
+    Log.d(LOG_TAG, "Got request to turn internet off, checking if it was present")
+    reachability.waitForNetworkStatus { it != NetworkStatus.NOT_REACHABLE }
+    val instrumentation = InstrumentationRegistry.getInstrumentation()
+    val uiAutomation = instrumentation.uiAutomation
+    Log.d(LOG_TAG, "turning off wifi and mobile data")
     uiAutomation.executeShellCommand("svc wifi disable")
     uiAutomation.executeShellCommand("svc data disable")
     try {
+        assumeNetworkIsNotReachable(reachability)
         block()
     } finally {
+        Log.d(LOG_TAG, "turning on wifi and mobile data")
         uiAutomation.executeShellCommand("svc wifi enable")
         uiAutomation.executeShellCommand("svc data enable")
     }
+    Log.d(LOG_TAG, "Waiting for network to become reachable")
+    reachability.waitForNetworkStatus { it != NetworkStatus.NOT_REACHABLE }
+}
+
+private suspend fun assumeNetworkIsNotReachable(reachability: ReachabilityInterface) {
+    val networkIsReachable = withTimeoutOrNull(3000) {
+        reachability.waitForNetworkStatus { it == NetworkStatus.NOT_REACHABLE }
+        false
+    } ?: true
+    assumeFalse(
+        "network should not be reachable if it's turned off on device",
+        networkIsReachable
+    )
```

Comment:
Here's how it looks on firebase for one device that doesn't let us turn its WiFi off:
<img width="1722" alt="image" src="https://github.com/mapbox/mapbox-navigation-android/assets/6190346/41ac887e-7f53-4a0f-8455-e6845614366e">